spy.py

```diff
@@ -0,0 +1,78 @@
+import libvirt
+import sys
+import json
+from flask import Flask, request, Response, make_response
+
+try:
+    conn = libvirt.open()
+except libvirt.libvirtError:
+    print('Failed to open connection to the hypervisor')
+    sys.exit(1)
+
+app = Flask(__name__)
+
+class JSONResponseEncoder(json.JSONEncoder):
+    """
+    BSON ObjectId-safe JSON response encoder
+    """
+
+    def default(self, o):
+        if isinstance(o, ObjectId):
+            return str(o)
+        return json.JSONEncoder.default(self, o)
+
+def _resp(success: bool, message: str, data: object = None):
+        """
+        Return a JSON API response
+        :param success: did the request succeed?
+        :param message: what happened?
+        :param data: any application specific data
+        """
+
+        return Response(JSONResponseEncoder().encode({
+            "meta": {
+                "success": success,
+                "message": message
+            },
+            "data": data
+        }), mimetype="application/json")
+
+def DomainToDict(vm):
+        vm_info = vm.info()
+        return {'name': vm.name(), 'state': vm_info[0], 'maxMemory': vm_info[1], 'vCPUs': vm_info[3]}
+
+
+@app.route("/vms")
+def listVMs():
+    try:  # getting a list of all domains (by ID) on the host
+        domains = conn.listAllDomains(0)
+    except:
+        raise Exception('Failed to find any domains')
+
+    vms = []
+    for vm in domains:
+        vms.append(DomainToDict(vm))
+
+    return _resp(True, "Virtual Machines", data=vms)
+
+@app.route("/vm/<name>/start")
+def startVM(name):
+    vm = conn.lookupByName(name)
+    try:
+        vm.create()
+        return _resp(True, "Started VM")
+    except Exception as e:
+        return _resp(False, str(e))
+
+@app.route("/vm/<name>/stop")
+def stopVM(name):
+    vm = conn.lookupByName(name)
+    try:
+        vm.shutdown()
+        return _resp(True, "Stopped VM")
+    except Exception as e:
+        return _resp(False, str(e))
+
+with open("/etc/wireguard/wg0.conf") as f:
+    content = f.readlines()
+    app.run(host=content[1].split(" = ")[1].split("/")[0])
```

Comment:
Not a bad idea, that's pretty creative to pull the listen address from the wg config. Lets add a comment here so we don't forget what this does later down the line